### PR TITLE
TYPE: Extend extract to support return parameter

### LINKED ISSUE #1774: Extracting a function should create the necessary parameters.
It would be nice if when extracting a function from code all the variables used by that range of code became parameters to the new function.

For example:

```rust
fn foo(x: u32) -> u32 {
   x *= 2;
   x += 1;
   x
}
```
became
```rust

fn foo(x: u32) -> u32 {
    x *= 2;
    new_fn(x)
}

fn new_fn(x: u32) -> u32 {
   x += 1
   x
}
```

### PR DESCRIPTION
**Pull request by farodin91**
Partial fixes #1774 
@vlad20012 r?

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt [hunk 1]
```diff
@@ -31,11 +38,24 @@ class RsExtractFunctionConfig private constructor(
             // check element should be a part of one block
             val fn = first.parentOfType<RsFunction>() ?: return null
             if (fn != last.parentOfType<RsFunction>()) return null
+
+            val letBindings = fn.descendantsOfType<RsPatBinding>().filter { it.textOffset <= end }
+
+            //val beforeBindings = letBindings.filter { it.textOffset < start }
```
Comment (matklad) [inline src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt]: This probably should be deleted?

File: src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionDialog.kt [hunk 2]
```diff
@@ -6,47 +6,76 @@
...
     }
     visibilityBox.selectedItem = "Private"
 
-    with(contentPanel) {
-        add(visibilityLabel)
-        add(functionNameLabel)
-        add(visibilityBox)
-        add(functionNameField)
+    val signatureComponent = RsSignatureComponent("fn test() {}", project)
+    signatureComponent.minimumSize = Dimension(300,30)
```
Comment (matklad) [inline src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionDialog.kt]: Formatting is off: there should be a space after `,`.

File: src/test/kotlin/org/rust/lang/refactoring/RsExtractFunctionTest.kt [hunk 3]
```diff
@@ -33,6 +33,48 @@ class RsExtractFunctionTest : RsTestBase() {
...
+    fun `test extract take last possible return binding`() = doTest("""
+            fn main() {
+                <selection>let test2 = 10i32;
+                let test = 10i32;</selection>
+                println!("{}", test);
+                println!("{}", test2);
+            }
+        """, """
+            fn main() {
+                let test = test();
```
Comment (matklad) [inline src/test/kotlin/org/rust/lang/refactoring/RsExtractFunctionTest.kt]: Yeah, ideally, we should be able to do something like `let (test, test2) = test()`...

File: src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt (hunk 1 above)
Comment (farodin91) [reply src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt]: This is prep for function parameters.

**Comment by farodin91** [comment]
@matklad all resolved

**Comment by matklad** [comment]
bors r+

Thanks @farodin91 !

**Pull request merged**